src/crimson/osd/pg.cc

```diff
@@ -1771,4 +1771,20 @@ PG::already_complete(const osd_reqid_t& reqid)
   }
 }
 
+void PG::remove_object_maybe_snapmapped(
+  ceph::os::Transaction &t,
+  const hobject_t &soid)
+{
+  t.remove(
+    coll_ref->get_cid(),
+    ghobject_t{soid, ghobject_t::NO_GEN, pg_whoami.shard});
+  if (soid.snap < CEPH_MAXSNAP) {
```

Comment:
I'm ok with uniting `remove_snap_mapped_object` and `clear_object_snap_mapping`. Naming suggestion below, up to you :)
```
PG::remove_snap_mapped_object(
  ..
  ..
  // clear object snap mapping
  if (soid.snap < CEPH_MAXSNAP) {
    ..   
```

Reply:
All done, I changed the name to `remove_maybe_snapmapped_object()` as the object being removed may not have snaps.